src/utils.ts

```diff
@@ -79,3 +80,21 @@ export namespace Option {
     return x !== undefined
   }
 }
+
+export function createLogger<T>(atom: Atom<T>, name?: string): Atom<T> {
+  let prevState = atom.get()
+
+  atom.subscribe(state => {
+    console.group(`UPDATE ${name ? `TYPE: ${name} ` : ''}@ ${new Date().toTimeString()}`)
+    log('prev state', '#9E9E9E', prevState)
+    log('next state', '#4CAF50', state)
+    console.groupEnd()
+    prevState = state
+  })
+
+  return atom
+
+  function log(msg: string, color: string, x: T) {
+    console.log('%c' + msg, `color: ${color}; font-weight: bold`, x)
+  }
```

Comment:
Let's move the definition for `log` to above `atom.subscribe` call, so that it is defined before it is used.